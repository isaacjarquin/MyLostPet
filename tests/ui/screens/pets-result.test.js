import "react-native"
import { ListView, ListViewDataSource } from "react-native"
import React from "react"
import PetsResult from "../../../screens/pets-result"

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import renderer from "react-test-renderer"
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe("PetsResult", () => {
    const navigate = jest.fn()

    const props = {
        navigation: {
            navigate: navigate,
            state: {
                params: {
                    pets: []
                }
            }
        }
    };

    beforeEach(() => {
        wrapper = shallow(<PetsResult {...props} />);
    })

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    describe("onLocationFocus", ()=> {
        it("modify the state when called", () => {
            wrapper.setState({ 
                locationFocusColor: "",
                breedFocusColor: ""
            })
            wrapper.instance().onLocationFocus()
            expect(wrapper.state().locationFocusColor).toBe("white")
            expect(wrapper.state().breedFocusColor).toBe("grey");
        })
    })

    describe("onBreedFocus", () => {
        it("modify the state when called", () => {
            wrapper.setState({
                breedFocusColor: "",
                locationFocusColor: ""
            })
            wrapper.instance().onBreedFocus()
            expect(wrapper.state().breedFocusColor).toBe("white")
            expect(wrapper.state().locationFocusColor).toBe("grey");
        })
    })

    describe("setAndfilterbyCity", () => {
        it("modify the state when called", () => {
            wrapper.setState({
                breed: "",
                location: ""
            })

            wrapper.instance().setAndfilterbyCity("location")
            expect(wrapper.state().location).toBe("location")
            expect(wrapper.state().breed).toBe("");
        })
    })

    describe("setAndfilterbyBreed", () => {
        it("modify the state when called", () => {
            wrapper.setState({
                breed: "",
                location: ""
            })

            wrapper.instance().setAndfilterbyBreed("breed")
            expect(wrapper.state().location).toBe("")
            expect(wrapper.state().breed).toBe("breed");
        })
    })

    describe("filterPets", () => {
        it("returns all items when no filter is selected", () => {
            const pet = {
                location: 'Tenerife',
                breed: 'Pastor aleman'
            }
            const pet1 = {
                location: 'Las palmas',
                breed: 'pitbull'
            }
            wrapper.setState({ pets: [pet, pet1] })
            
            wrapper.instance().filterPets({ location: "", breed: "" })
            expect(wrapper.state().dataSource.getRowCount()).toBe(2)
        })

        it("returns one items when only one item match the filter", () => {
            const pet = {
                location: 'Tenerife',
                breed: 'Pastor aleman'
            }
            const pet1 = {
                location: 'Las palmas',
                breed: 'pitbull'
            }
            wrapper.setState({ pets: [pet, pet1] })

            wrapper.instance().filterPets({ location: "Las palmas", breed: "" })
            expect(wrapper.state().dataSource.getRowCount()).toBe(1)
        })

        it("returns no items when search doesnt match the filter", () => {
            const pet = {
                location: 'Tenerife',
                breed: 'Pastor aleman'
            }
            const pet1 = {
                location: 'Las palmas',
                breed: 'pitbull'
            }
            wrapper.setState({ pets: [pet, pet1] })

            wrapper.instance().filterPets({ location: "Las palmas", breed: "Pastor aleman" })
            expect(wrapper.state().dataSource.getRowCount()).toBe(0)
        })
    })
})
