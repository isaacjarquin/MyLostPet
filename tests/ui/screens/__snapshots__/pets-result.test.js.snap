// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#333333",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <RCTScrollView>
    <View>
      <RCTScrollView
        dataSource={
          ListViewDataSource {
            "items": 1,
          }
        }
        enableEmptySections={true}
        renderRow={[Function]}
        renderScrollComponent={[Function]}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#bbb",
                    "borderBottomWidth": 1,
                    "paddingBottom": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "height": 90,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                  },
                ],
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 34,
                    "width": 34,
                  }
                }
              >
                <View
                  activeOpacity={undefined}
                  onLongPress={undefined}
                  onPress={undefined}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "height": 34,
                        "paddingBottom": 10,
                        "paddingRight": 10,
                        "paddingTop": 10,
                        "width": 34,
                      },
                      Object {
                        "borderRadius": 17,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "backgroundColor": "rgba(0,0,0,0.2)",
                          "bottom": 0,
                          "flex": 1,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "borderRadius": 17,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "imageupload/w_100,h_100,c_fill,g_southundefined",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "height": 34,
                            "width": 34,
                          },
                          Object {
                            "borderRadius": 17,
                          },
                          Object {
                            "borderRadius": 32,
                            "height": 64,
                            "marginLeft": 10,
                            "marginRight": 10,
                            "marginTop": 35,
                            "width": 64,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginBottom": 5,
                      "marginLeft": 15,
                      "marginTop": 4,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontSize": 17.5,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          Object {
                            "color": "white",
                            "fontWeight": "bold",
                            "opacity": 0.9,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    kind, de raza breed
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "marginLeft": 15,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={2}
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#86939e",
                            "fontSize": 15,
                            "fontWeight": "600",
                            "marginTop": 1,
                          },
                          Object {
                            "marginLeft": 10,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    Encontrado en location, el date. info
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flex": 0.15,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#bdc6cf",
                          "fontSize": 28,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        onPress={undefined}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "height": 70,
            "opacity": 0.8,
            "paddingLeft": 12,
            "paddingTop": 22,
            "width": 50,
          }
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 25,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        color="grey"
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Ciudad/Municipio..."
        placeholderTextColor="grey"
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "fontSize": 18,
            "height": 70,
            "opacity": 0.8,
            "padding": 10,
            "paddingLeft": 0,
            "width": 650,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        onPress={undefined}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "height": 70,
            "opacity": 0.8,
            "paddingTop": 20,
            "width": 50,
          }
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 25,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        onPress={undefined}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "height": 70,
            "opacity": 0.8,
            "paddingLeft": 12,
            "paddingTop": 22,
            "width": 50,
          }
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 25,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <TextInput
        color="grey"
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Raza..."
        placeholderTextColor="grey"
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "fontSize": 18,
            "height": 70,
            "opacity": 0.8,
            "padding": 10,
            "paddingLeft": 0,
            "width": 650,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        onPress={undefined}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderTopWidth": 1,
            "height": 70,
            "opacity": 0.8,
            "paddingTop": 20,
            "width": 50,
          }
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 25,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
